internal/handler/Handler.go

```diff
@@ -0,0 +1,323 @@
+		return
+	}
+	_, er := w.Write(jsonBytes)
+	if er != nil {
+		http.Error(w, "error with server", http.StatusInternalServerError)
+	}
+	w.WriteHeader(http.StatusOK)
+}
+
+func (h *Handler) getEmployees(w http.ResponseWriter, r *http.Request) {
+	query, err := url.ParseQuery(r.URL.RawQuery)
+	limit, err := strconv.ParseInt(query["limit"][0], 10, 64)
+	if err != nil {
+		http.Error(w, "error with server", http.StatusInternalServerError)
+		return
+	}
+	offset, err := strconv.ParseInt(query["offset"][0], 10, 64)
+	if err != nil {
+		http.Error(w, "error with server", http.StatusInternalServerError)
+		return
+	}
+	employees, err := h.service.GetEmployees(int(limit), int(offset))
+	if err != nil {
+		if err.Error() == "incorrect data"{
+			http.Error(w, "error: not found", http.StatusNotFound)
+			return
+		}else {
+			http.Error(w, "error with server", http.StatusInternalServerError)
+			return
+		}
+	}
+	jsonBytes, err := json.Marshal(employees)
+	if err != nil {
+		http.Error(w, "error with server", http.StatusInternalServerError)
+		return
+	}
+	_, er := w.Write(jsonBytes)
+	if er != nil {
+		http.Error(w, "error with server", http.StatusInternalServerError)
+	}
+	w.WriteHeader(http.StatusOK)
+}
+
+func (h *Handler) getPosition(w http.ResponseWriter, r *http.Request) {
+	matches := positionRe.FindStringSubmatch(r.URL.Path)
+	if len(matches) < 2 {
+		http.NotFound(w, r)
+		return
+	}
+	p, err := h.service.GetPosition(matches[1])
+	if err != nil {
+		w.WriteHeader(http.StatusNotFound)
+		_, err := w.Write([]byte("Position not found"))
+		if err != nil {
+			http.Error(w, "error: bad request", http.StatusBadRequest)
+		}
+		return
+	}
+	jsonBytes, err := json.Marshal(p)
+	if err != nil {
+		http.Error(w, "error with server", http.StatusInternalServerError)
+		return
+	}
+	_, er := w.Write(jsonBytes)
+	if er != nil {
+		http.Error(w, "error: with server", http.StatusInternalServerError)
+	}
+	w.WriteHeader(http.StatusOK)
+}
+
+func (h *Handler) getEmployee(w http.ResponseWriter, r *http.Request) {
+	matches := employeeRe.FindStringSubmatch(r.URL.Path)
+	if len(matches) < 2 {
+		http.NotFound(w, r)
+		return
+	}
+	e, err := h.service.GetEmployee(matches[1])
+	if err != nil {
+		w.WriteHeader(http.StatusNotFound)
+		_, err := w.Write([]byte("employee not found"))
+		if err != nil {
+			http.Error(w, "error: bad request", http.StatusBadRequest)
+		}
+		return
+	}
+	jsonBytes, err := json.Marshal(e)
+	if err != nil {
+		http.Error(w, "error with server", http.StatusInternalServerError)
+		return
+	}
+	_, er := w.Write(jsonBytes)
+	if er != nil {
+		http.Error(w, "error: with server", http.StatusInternalServerError)
+	}
+	w.WriteHeader(http.StatusOK)
+}
+
+func (h *Handler) createPosition(w http.ResponseWriter, r *http.Request) {
+	var p internal.Position
+	if err := json.NewDecoder(r.Body).Decode(&p); err != nil {
+		http.Error(w, "error with server", http.StatusInternalServerError)
+		return
+	}
+	err := h.service.CreatePosition(p)
+	if err != nil {
```

Comment:
The same comment here as I've left on getPosition() handler. You should return a special error in case if position already exists  and then return corresponding error - 404 or 505 - if there is any other error.

Reply:
+ the same for createEmployee() handler

Reply:
+